shell/platform/android/android_shell_holder.h

```diff
@@ -46,12 +112,17 @@ class AndroidShellHolder {
   const flutter::Settings settings_;
   const std::shared_ptr<PlatformViewAndroidJNI> jni_facade_;
   fml::WeakPtr<PlatformViewAndroid> platform_view_;
-  ThreadHost thread_host_;
+  std::shared_ptr<ThreadHost> thread_host_;
   std::unique_ptr<Shell> shell_;
   bool is_valid_ = false;
   uint64_t next_pointer_flow_id_ = 0;
+  std::shared_ptr<AssetManager> asset_manager_;
 
   static void ThreadDestructCallback(void* value);
+  std::optional<RunConfiguration> BuildRunConfiguration(
+      std::shared_ptr<flutter::AssetManager> asset_manager,
+      std::string entrypoint,
```

Comment:
nit: `const std::string&` to avoid unnecessary copies

Reply:
thanks! Changed Launch too